code/modules/power/supermatter/supermatter.dm

```diff
@@ -413,21 +420,41 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 		bzcomp += clamp(max(removed.gases[/datum/gas/bz][MOLES]/combined_gas, 0) - bzcomp, -1, gas_change_rate)
 		n2ocomp += clamp(max(removed.gases[/datum/gas/nitrous_oxide][MOLES]/combined_gas, 0) - n2ocomp, -1, gas_change_rate)
 		n2comp += clamp(max(removed.gases[/datum/gas/nitrogen][MOLES]/combined_gas, 0) - n2comp, -1, gas_change_rate)
+		h2ocomp += clamp(max(removed.gases[/datum/gas/water_vapor][MOLES]/combined_gas, 0) - h2ocomp, -1, gas_change_rate)
 
 		//We're concerned about pluoxium being too easy to abuse at low percents, so we make sure there's a substantial amount.
 		if(pluoxiumcomp >= 0.15)
 			pluoxiumbonus = 1    //makes pluoxium only work at 15%+
 		else
 			pluoxiumbonus = 0
 
+		if (h2ocomp >= 0.35 && h2ocomp <= 0.45) //heat protection from h2o only works when it's between 35 and 45%
+			h2obonus = 1
+		else
+			h2obonus = 0
+
+		if (h2ocomp >= 0.15 && h2ocomp <= 0.45) //the engine will stop producing power from 15% to 45% h2o (slow down the o2 and plasma generation too)
+			h2omalus = 0.05
+		else if (h2ocomp < 0.05)
+			h2omalus = 1
+		else
+			h2omalus = 2  //when gas comp is above 0.45 or between 0.15 and 0.05, the engine will start to freak out (may need test on the amount of freak out)
+
+		if (h2ocomp < 0.30) //variability in the h2o penalty calculation depending on the %
+			h2ofixed = 1
+		else if (h2ocomp <= 0.50)
+			h2ofixed = 0.5
+		else
+			h2ofixed = 2
```

Comment:
You can use a switch to act on ranges
```
switch(h2ocomp)
  if(-INFINITY to 0.3)
    ...
  if(0.3 to 0.5)
    ...
  if(0.5 to INFINITY)
    ...
```

Reply:
i'll do it

Reply:
Check that the -INFINITY works in the switch, that has caused issues in the past for me.

Reply:
It does

Reply:
done